spec/killbill_client/encoding_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe KillBillClient::API do
+
+  it 'should send double-encoded uri' do
+    plugin_properties = [
+        {
+            'key' => 'contractId',
+            'value' => "test"
+        },
+        {
+            'key'   => 'details',
+            'value' =>  {
+                'eventType' => 'voidEvent',
+                'transactionType' => 'void',
+                'contractType' => 'temp'
+            }.to_json
+        }
+    ].map{|hash| KVPair.from_hash(hash)}
```

Comment:
@pierre Add KVPair is to be able to visit key and value as called in https://github.com/killbill/killbill-client-ruby/blob/master/lib/killbill_client/api/net_http_adapter.rb#L53

Do you have any suggestions on how to simply this?